### ISSUE TITLE
`AttributeError: 'NoneType' object has no attribute 'return_type'` on connection fields

### ISSUE THREAD

**Issue by LucasPickering** [blb-ventures/strawberry-django-plus#134]
I have a couple top-level connection fields that, when queried, get this error:

```
Traceback (most recent call last):
  File "/root/.cache/pypoetry/virtualenvs/beta-spray-9TtSrW0h-py3.10/lib/python3.10/site-packages/graphql/execution/execute.py", line 521, in execute_field
    result = resolve_fn(source, info, **args)
  File "/root/.cache/pypoetry/virtualenvs/beta-spray-9TtSrW0h-py3.10/lib/python3.10/site-packages/strawberry_django_plus/optimizer.py", line 605, in resolve
    ret = _next(root, info, *args, **kwargs)
  File "/root/.cache/pypoetry/virtualenvs/beta-spray-9TtSrW0h-py3.10/lib/python3.10/site-packages/strawberry/schema/schema_converter.py", line 421, in _get_basic_result
    return field.get_result(_source, info=None, args=[], kwargs={})
  File "/root/.cache/pypoetry/virtualenvs/beta-spray-9TtSrW0h-py3.10/lib/python3.10/site-packages/strawberry_django_plus/relay.py", line 984, in get_result
    type_def = info.return_type._type_definition  # type:ignore
AttributeError: 'NoneType' object has no attribute 'return_type'
```

This is on version `1.28.3`. After doing some digging, I believe it relates to [this commit on strawberry](https://github.com/strawberry-graphql/strawberry/commit/f14d92362d9db3cd0955ce49d1310f224dd67ab9). I added some debug logging, and the `is_basic_field` property added in that commit returns true for the top-level connection fields, as they have no `base_resolver` defined. Interestingly, the `base_resolver` _is_ defined for a nested connection field, so that one works correctly. That's where I gave up on debugging -- I'm not sure what the `base_resolver` is or where it comes from.

**Comment by bellini666** [comment]
Hey @LucasPickering ,

Are you using the latest version of this lib? Because it already has a fix for that change from strawberry: https://github.com/blb-ventures/strawberry-django-plus/commit/fd8ca293560d6c7f34b7236d41ac2ced065b6a85

**Comment by LucasPickering** [comment]
> Hey @LucasPickering ,
> 
> Are you using the latest version of this lib? Because it already has a fix for that change from strawberry: [fd8ca29](https://github.com/blb-ventures/strawberry-django-plus/commit/fd8ca293560d6c7f34b7236d41ac2ced065b6a85)

Yep, I just double checked that I'm on `1.28.3`. I also checked the source code for the library I have installed and confirmed that the changes from that commit are present locally for me. I noticed that `ConnectionField` extends `RelayField`, but that commit added `is_basic_field` just to `NodeField`. Is that a bug? If so I'm happy to make a PR to move it to `RelayField` so it will cover connections too.
